odl/discr/grid.py

```diff
@@ -392,13 +388,11 @@ def stride(self):
         >>> rg.stride
         array([ 1.,  2.])
         """
-        if not self.is_uniform:
-            raise NotImplementedError('`stride` not defined for non-uniform '
-                                      'grid')
         strd = np.zeros(self.ndim)
-        strd[self.nondegen_byaxis] = (
-            self.extent[self.nondegen_byaxis] /
-            (np.array(self.shape)[self.nondegen_byaxis] - 1))
+        uniform = np.array(self.is_uniform_byaxis, dtype=bool)
+        strd[~uniform] = float('nan')
```

Comment:
Well, for the `strides` property it's a way of signalizing "not defined in this axis". Later on the NaN values are not processed since the non-uniform axes are not considered. At least if there's no bug in the code. Hopefully the NaNs will mess up later code that tries to wrongly process the non-uniform axes.